### DISCUSSION TITLE
Use TCPserial instead of bluetooth

### DISCUSSION THREAD

**Discussion by pedro042** [fl4p/batmon-ha#72] [Q&A]
Hi,
thank you for this nice software. Currently I've JBD BMS, but I'm going to replace it with JK. Tried several times this software. It works well for me, but bluetooth is not reliable (maybe I've buggy BT stick). I bought PUSR N540 device, it's in short converter from TCP to RS485 serial, 4 RS-485 ports. Currently I use it for Epever inverter and Epever MPPT controler. The communication is very reliable and stable.
I'd like to use it for communication with BMS too (BMS has RS-485 port). Can anyone suggest if it's possible to use TCP instead of bluetooth with this software (similar to jbdtool, jktool etc).

Some advantages:
- no distance limit
- reliability
- BT attacks not possible
- and probably more
Thanks

**Comment by ttiburst** [comment]
do you get an answer 
i have the same cocern

**Reply by pedro042** [reply]
no, not resolved. I decided to use ESP32 with ethernet port and esphome-jkbms and install the ESP32 board close to the BMS. I hope 10 cm is OK ;) for reliable bluetooth communication. Computer connected via ethernet, no wifi.
I'll try to connect JK BMS serial port to the ESP32 board directly, but I'm not sure if all features are supported via serial port (because of lack of documentation)
